torch/testing/_asserts.py

```diff
@@ -114,24 +184,41 @@ def _check_supported_tensor(
     """
     if input.is_quantized:
         return UsageError("Comparison for quantized tensors is not supported yet.")
-    if input.is_sparse:
-        return UsageError("Comparison for sparse tensors is not supported yet.")
 
     return None
 
 
+class _TensorFormat:
+    def __init__(self, t: torch.Tensor) -> None:
+        self.is_sparse, self.sparse_format = self._extract_format(t)
+
+    @staticmethod
+    def _extract_format(t: torch.Tensor) -> Tuple[bool, Optional[str]]:
+        if t.is_sparse:
+            return True, "COO"
+        elif t.is_sparse_csr:
+            return True, "CSR"
+        else:
+            return False, None
```

Comment:
@pmeier I think the @mruberry concern about what happens when introducing a new sparse format is related to getting a sensible error message from the `assert_close` framework that it needs to be updated to the new sparse format.
Here, for instance, returning `False` may give a false-negative result for `is_sparse` attribute (e.g. when a new sparse format is introduced), and what follows will be undefined behaviour.

I would recommend being explicit in strided layout checks and raise exceptions as early as possible when something needs to be updated for new features.